py/desispec/scripts/humidity_corrected_fiberflat.py

```diff
@@ -0,0 +1,101 @@
+
+from __future__ import absolute_import, division
+
+import os
+import fitsio
+import argparse
+import numpy as np
+
+from desiutil.log import get_logger
+
+from desispec.io import read_fiberflat,write_fiberflat,findfile,read_frame
+from desispec.io.fiberflat_vs_humidity import get_humidity,read_fiberflat_vs_humidity
+from desispec.calibfinder import CalibFinder
+from desispec.fiberflat_vs_humidity import compute_humidity_corrected_fiberflat
+
+def parse(options=None):
+    parser = argparse.ArgumentParser(description="Compute a fiberflat corrected for variations with humidity.")
+
+    parser.add_argument('-i','--infile', type = str, default = None, required=True,
+                        help = 'path of DESI exposure frame fits file')
+    parser.add_argument('--fiberflat', type = str, default = None, required=True,
+                        help = 'path of DESI fiberflat fits file')
+    parser.add_argument('--use-sky-fibers', action = 'store_true',
+                        help = 'use sky fibers to improve the correction')
+    parser.add_argument('-o','--outfile', type = str, default = None, required=True,
+                        help = 'path of output fiberflar file')
+    args = None
+    if options is None:
+        args = parser.parse_args()
+    else:
+        args = parser.parse_args(options)
+    return args
+
+
+def main(args) :
+
+    log = get_logger()
+
+    # just read frame header in case we don't need to do anything
+    frame_header = fitsio.read_header(args.infile,"FLUX")
+
+    if args.use_sky_fibers :
+        # need full frame to adjust correction on data
+        frame = read_frame(args.infile)
+    else :
+        frame = None
+
+    cfinder = CalibFinder([frame_header])
+    if not cfinder.haskey("FIBERFLATVSHUMIDITY"):
+        log.info("No information on fiberflat vs humidity for camera {}, simply link the input fiberflat".format(frame_header["CAMERA"]))
+        if not os.path.islink(args.outfile) :
+            os.symlink(args.fiberflat,args.outfile)
+        return 0
+
+    # read fiberflat
+    calib_fiberflat = read_fiberflat(args.fiberflat)
+
+    # read mean fiberflat vs humidity
+    filename = cfinder.findfile("FIBERFLATVSHUMIDITY")
```

Comment:
in $DESI_SPECTRO_CALIB, let's include a date in the name of the fiberflat-vs-humidity-*.fits files to establish a naming scheme that naturally supports having having multiple versions.  e.g. in the future we'll need a different correction file after the collimators are replaced.  I know the yaml structure allows the names to be arbitrary, but let's not have the first one be the one without a date.

Reply:
done